cli/JSXV4.md

```diff
@@ -76,15 +76,95 @@ module Wrapper = {
 let make = () => body
 ```
 
-**V4 Spec**
+### Migration of V3 components that depend on the internal representation
+
+Some components in existing projects are written in a way that is dependent on the V3 internal representation.
+Here are a few examples of how to convert them to V4.
+
+#### `makeProps` does not exist in V4
+
+Rewrite this:
+
+```rescript
+// V3
+module M = {
+  @obj external makeProps: (~msg: 'msg, ~key: string=?, unit) => {"msg": 'msg} = "" // No more makeProps
+
+  let make = (~msg) => {
+    <div> {React.string(msg)} </div>
+  }
+}
+```
+
+To this:
+
+```rescript
+// V4
+module M = {
+  type props<'msg> = {msg: 'msg}
+  let make = props => <div> {React.string(props.msg)} </div>
+}
+```
+
+#### `React.Context`
```

Comment:
@mattdamon108 would you clarify this example?
Is it necessary to depend on the internal representation in order to use `React.Context`?
If not, why is the fix not to avoid using the internal representation?

Reply:
I saw the error report from the community and the official doc shows the example of using internal representation to use `React.Context`. Therefore, many users might use this code and are expected to face the error with v3.

Reply:
@mattdamon108 what's the answer to my question?
Should the solution in V4 use the V4 runtime representation, or should it not depend on the runtime representation?

Reply:
> @mattdamon108 what's the answer to my question? Should the solution in V4 use the V4 runtime representation, or should it not depend on the runtime representation?

The solution should use the v4 runtime representation.

Reply:
> > @mattdamon108 what's the answer to my question? Should the solution in V4 use the V4 runtime representation, or should it not depend on the runtime representation?
> 
> The solution should use the v4 runtime representation.

How about this instead:
```res
    module Prov = {
      let make = React.Context.provider(context)
    }
    ...
    <Prov value children /> // OK
```

Does it work?

Reply:
Let me test it.

Reply:
> Does it work?

Wow, it works. But, I'm bit unsure, because the js output different. I prefer and think the js output with internal representation is fine. But yes it works.

Reply:
> > Does it work?
> 
> Wow, it works. But, I'm bit unsure, because the js output different. I prefer and think the js output with internal representation is fine. But yes it works without it.

Sounds good.
I will let you have one final pass to the examples, and then will merge.
OK?

Reply:
https://github.com/rescript-lang/syntax/pull/652/commits/22ec394d56b800b3446af0f6a33334fa4625f6f8